### PR TITLE
Resolve #15: Update from older version of p4java-jfrog to latest versjon...

### LINKED ISSUE #15: Upgrade to latest release of p4java
The gradle-info-plugin currently uses p4java version 2011.1.297684 (com.perforce:p4java-jfrog:2011.1.297684), which requires com.jcraft:jzlib:1.0.7.

This version of jzlib conflicts with jRuby, which in turns is used by asciidoctor, and which ultimately means that there are problems using both the info plugin and asciidoctor-pdf in the same gradle build. See https://github.com/elygre/asciidoctor-pdf-final-class and http://discuss.asciidoctor.org/Using-asciidoctorj-pdf-from-gradle-tp2731p2986.html for more information on the conflict.

This issue is a request to upgrade to the latest release of p4java 2014.1.965322, ref https://bintray.com/bintray/jcenter/com.perforce%3Ap4java/2014.1.965322/view. This version also fixes the INDEX.LIST-issue that forced the use of a third party packaging of p4java, ref https://github.com/nebula-plugins/gradle-info-plugin/blob/gradle-2.2/build.gradle#L19-L24.

I do not have access to a perforce repository anywhere, so I am unable to properly test and verify this change. However, the following change in build.gradle builds successfully:

    -    compile 'com.perforce:p4java-jfrog:2011.1.297684'
    +    compile 'com.perforce:p4java:2014.1.965322'

### PR DESCRIPTION
**Pull request by elygre**
... of p4java, which does not have the problem that prompted the use of p4java-jfrog in the first place.

Note that I do not have access to a perforce repository, so testing has been limited to compiling and building (including running the project's unit tests).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -16,12 +16,7 @@ contacts {
 dependencies {
     // TODO, make these optional
 
-    // Specifically not using com.perforce:p4java since it has a META-INF/INDEX.LIST file that lists
-    // com.jcraft. In additional to wrongness of bundle thirdparty libraries, it doesn't even have 
-    // those class files. The classloader will consider the INDEX.LIST file depending on ordering
-    // of the classpath, so it's very likely that just havine p4java in your path will prevent the
-    // JVM from even running.
-    compile 'com.perforce:p4java-jfrog:2011.1.297684'
+    compile 'com.perforce:p4java:2014.1.965322'
```
Comment (bmuschko) [inline build.gradle]: Seems like there was a reason for not using `p4java` and instead using `p4java-jfrog`. @quidryan Can you comment on this pull request?

Comment (elygre) [inline build.gradle]: A comment from the original committer would be great.

The INDEX.LIST file is gone from the latest release though, as mentioned in the description of issue #15:

>This version also fixes the INDEX.LIST-issue that forced the use of a third party packaging of p4java, ref https://github.com/nebula-plugins/gradle-info-plugin/blob/gradle-2.2/build.gradle#L19-L24.

**Comment by quidryan** [comment]
As long as the latest p4java is missing INDEX.LIST, it should be fine to pull in.

**Comment by elygre** [comment]
Just realized that this merge request had its first birthday recently. Any chance of merging?

**Comment by DanielThomas** [comment]
🎉 

(I'll close this without merging and upgrade to the latest version instead.)

**Comment by DanielThomas** [comment]
Included in v3.1.0.

**Pull request closed without merging**